CRM/Contribute/Tokens.php

```diff
@@ -47,61 +33,65 @@ protected function getApiEntityName(): string {
   }
 
   /**
-   * Get a list of tokens for the entity for which access is permitted to.
+   * Get a list of simple, passthrough tokens which are loaded via APIv4.
    *
    * This list is historical and we need to question whether we
    * should filter out any fields (other than those fields, like api_key
    * on the contact entity) with permissions defined.
    *
    * @return array
+   *   Ex: ['foo' => 'Foo Stuff', 'bar_id' => 'Barber ID#']
    */
-  protected function getExposedFields(): array {
-    $fields = [
-      'contribution_page_id',
-      'source',
-      'id',
-      'receive_date',
-      'total_amount',
-      'fee_amount',
-      'net_amount',
-      'non_deductible_amount',
-      'trxn_id',
-      'invoice_id',
-      'currency',
-      'cancel_date',
-      'receipt_date',
-      'thankyou_date',
-      'tax_amount',
-      'contribution_status_id',
-      'financial_type_id',
-      'payment_instrument_id',
-      'cancel_reason',
-      'amount_level',
-      'check_number',
-    ];
+  protected function getApiTokens(): array {
+    $result = parent::getApiTokens();
```

Comment:
Sounds like #21184 provides the list of PC types. Then the parent implementation of `getApiTokens()` can provide the `:name` (etc). So this override goes away.